src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,171 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int currentSize;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (currentSize == 0) {
+            addFirstNode(value);
+            return;
+        }
+        Node<T> node = new Node<>(tail, value, null);
+        tail.next = node;
+        tail = node;
+        currentSize++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > currentSize) {
+            throw new IndexOutOfBoundsException();
```

Comment:
When you throw exeption write description as:
throw new IndexOutOfBoundsException("Index " + index + " out of bounds");